content/blog/2020-12-23-hof/index.md

```diff
@@ -0,0 +1,83 @@
++++
+title = """Higher-Order Functions in Bril"""
+[extra]
+bio = """
+Priya Srikumar is a senior undergraduate at Cornell University. They like researching compilers, verification, and programming languages!"""
+[[extra.authors]]
+name = "Priya Srikumar"
+link = "priyasrikumar.github.io"
++++
+
+## What was the goal? 
+
+I was aiming to implement higher-order functions (HOFs) in Bril by implementing a lambda lifter (which makes higher-order functions independently defined from each other globally) and adding support for local function definitions into the Bril syntax. I was also hoping to have functions as function arguments and return types, but this turned out to be more difficult than anticipated.
+
+## What did I do? 
+
+Everything I set out to do except for functions as function arguments and returning functions. 
+
+### Design
+
+Bril is an instruction-based language, so it was tricky to think about what functions would look like. I decided to allow variables to be instantiated with an `anon`
+
+#### Types
+
+I created a new parametrized `fun` type with the following syntax: 
+
+```
+{
+    "fun": {
+        "params" : [<Type>*]?, 
+        "ret" : [<Type>]?
+    }
+}
+```
+Both the `params` and the `ret` field are optional. 
+
+#### Operations 
+
+`anon` : Indicates a function definition. The first argument is a comma-separated parenthesis list of argument names. The second argument is a bracketed list of instructions and labels (like a normal function definition). 
+
+`apply` : Indicates a function application. The first argument is the function name. The subsequent arguments are arguments to that function.
+
+Here's an example of those two operations being used in a Bril program: 
```

Comment:
Maybe just clarify that you're switching to the text format rather than the JSON format for this example?